src/Microsoft.IdentityModel.Logging/LogHelper.cs

```diff
@@ -36,20 +37,109 @@ namespace Microsoft.IdentityModel.Logging
         {
-            IdentityModelEventSource.Logger.Write(logLevel, message, innerException);
+            return LogException<T>(EventLevel.Error, innerException, format, args);
+        }
+
+        /// <summary>
+        /// Logs an event using the event source logger and returns new typed exception.
+        /// </summary>
+        /// <param name="eventLevel">Identifies the level of an event to be logged.</param>
+        /// <param name="message">message to log.</param>
+        public static T LogException<T>(EventLevel eventLevel, string message) where T : Exception
+        {
+            return LogException<T>(eventLevel, null, message, null);
+        }
+
+        /// <summary>
+        /// Logs an event using the event source logger and returns new typed exception.
+        /// </summary>
+        /// <param name="eventLevel">Identifies the level of an event to be logged.</param>
+        /// <param name="format">Format string of the log message.</param>
+        /// <param name="args">An object array that contains zero or more objects to format.</param>
+        public static T LogException<T>(EventLevel eventLevel, string format, params object[] args) where T : Exception
+        {
+            return LogException<T>(eventLevel, null, format, args);
+        }
+
+        /// <summary>
+        /// Logs an event using the event source logger and returns new typed exception.
+        /// </summary>
+        /// <param name="eventLevel">Identifies the level of an event to be logged. Default is Error.</param>
+        /// <param name="innerException">the inner <see cref="Exception"/> to be added to the outer exception.</param>
+        /// <param name="message">message to log.</param>
+        public static T LogException<T>(EventLevel eventLevel, Exception innerException, string message) where T : Exception
+        {
+            return LogException<T>(eventLevel, innerException, message, null);
+        }
+
+        /// <summary>
+        /// Logs an event using the event source logger and returns new typed exception.
+        /// </summary>
+        /// <param name="eventLevel">Identifies the level of an event to be logged. Default is Error.</param>
+        /// <param name="innerException">the inner <see cref="Exception"/> to be added to the outer exception.</param>
+        /// <param name="format">Format string of the log message.</param>
+        /// <param name="args">An object array that contains zero or more objects to format.</param>
+        public static T LogException<T>(EventLevel eventLevel, Exception innerException, string format, params object[] args) where T : Exception
+        {
+            string message = null;
+
+            if (args != null)
+                message = string.Format(CultureInfo.InvariantCulture, format, args);
+            else
+                message = format;
+
+            if (IdentityModelEventSource.Logger.IsEnabled() && IdentityModelEventSource.Logger.LogLevel >= eventLevel)
```

Comment:
You are correct, I am being overly careful. Just in case the format string has params and the args is null, we would throw. This check sidesteps that possibility.